### PR TITLE
Cleanup Intellij Warnings

### PR DESCRIPTION
**Pull request by vkostyukov**
Taking an advatage of using Intellij IDEA and cleaning up some warinings:

* Unnecessary parentheses
* Anonynous function convertible to a method
* Unnecessary `final`
* Unused imports

### CODE REVIEW

File: core/src/main/scala/io/jfc/CursorOp.scala [hunk 1]
```diff
@@ -24,9 +24,9 @@ case class El(o: CursorOpElement, succeeded: Boolean) extends CursorOp {
 }
 
 object CursorOp {
-  def apply(o: CursorOpElement): CursorOp = El(o, true)
+  def apply(o: CursorOpElement): CursorOp = El(o, succeeded = true)
```
Comment (travisbrown) [inline core/src/main/scala/io/jfc/CursorOp.scala]: What's the motivation for these?

**Comment by travisbrown** [comment]
Thanks! This looks good to me, with the exception of the introduction of named parameters in a few places where I don't think they add much.

File: core/src/main/scala/io/jfc/Printer.scala [hunk 2]
```diff
@@ -129,13 +129,13 @@ final case class Printer(
       jsonString.span(Printer.isNormalChar) match {
         case (prefix, suffix) =>
           builder.append(prefix)
-          if (suffix.nonEmpty) appendJsonString(suffix, false)
+          if (suffix.nonEmpty) appendJsonString(suffix, normalChars = false)
```
Comment (travisbrown) [inline core/src/main/scala/io/jfc/Printer.scala]: Oh, so it looks like it just does these for booleans? I guess it kind of makes sense for clarity…

**Comment by travisbrown** [comment]
Okay, I'm sold on the named parameter for booleans—will merge when green. Thanks!

**Comment by vkostyukov** [comment]
Wait. I'm actually not sure we need them. Well Intellij suggests to use them for Boolean params, since it's not clear what `true` or `false` means without looking at the  function signature. But I recall some discussions about named params in Scala. As far as I remember, they considered "inpure".

**Comment by travisbrown** [comment]
I wouldn't want an API that was painful without named parameter, but I'm fine with this kind of usage—the reasoning makes sense to me.

**Pull request merged**